spec/runners/unified/test.rb

```diff
@@ -333,8 +333,10 @@ def execute_operation(op)
         else
           result = send(method_name, op)
           if expected_result = op.use('expectResult')
-            if result.nil? && !expected_result.empty?
-              raise Error::ResultMismatch, "Actual result nil but expected result #{expected_result}"
+            if result.nil? && expected_result.key?("$$unsetOrMatches")
```

Comment:
This needs to assert that unsetormatches value is truthy.

Reply:
Again I'm not sure that's correct as per my comments above. I think we might have different understandings about how unsetOrMatches works.